### PR TITLE
csds client: add filter support to node id

### PR DESCRIPTION
**Pull request by Rainton**
This PR includes adding flag **-filter_mode** and **-filter_pattern** to support filter on node_id. #34 

Filter modes currently supported:

- prefix
- suffix
- regex

Signed-off-by: Yutong Li yutongli@google.com
/cc @fuqianggao

### CODE REVIEW

File: csds-client/client/v2/client_test.go [hunk 1]
```diff
@@ -162,11 +162,98 @@ func TestVisualization(t *testing.T) {
...
+	var response csdspb_v2.ClientStatusResponse
+	if err = protojson.Unmarshal(responsejson, &response); err != nil {
+		t.Errorf("Read From File Failure: %v", err)
+	}
+	out := clientUtil.CaptureOutput(func() {
+		if err := printOutResponse(&response, c.opts); err != nil {
+			t.Errorf("Print out response error: %v", err)
+		}
+	})
+	want := "Client ID                                          xDS stream type                Config Status                  \ntest_node_1                                        test_stream_type1              N/A                            \ntest_node_2                                        test_stream_type2              N/A                            \ntest_node_3                                        test_stream_type3              N/A                            \n"
```
Comment (fuqianggao) [inline csds-client/client/v2/client_test.go:197]: can this be broken down to multiple lines? same elsewhere.

File: csds-client/README.md [hunk 2]
```diff
@@ -64,6 +64,10 @@ Common options are exposed/controlled via command line flags, while control plan
    * If the browser fails to open due to os version issue, you can copy the content in `config_graph.dot`, and then paste it in the edit box on the left of [Graphviz Online](https://dreampuf.github.io/GraphvizOnline/) or any other tools for [Graphviz](https://graphviz.org/) to show the graph of the dot file.
    * Each xDS node shown in the graph is labelled by index (e.g. LDS0, RDS0, RDS1,...) to make the graph more clear. The real name of xDS resource in config will show when the user hovers the mouse over each node.
    * If **the visualization mode** and **the monitor mode** are enabled together, the client will only save graph dot data for the latest response without opening the browser to avoid frequent pop-ups of the browser due to short monitor interval.
+* ***-filter_mode***: the filter mode for the filter on xDS nodes to be returned (e.g. prefix, suffix, regex, ...)
```
Comment (fuqianggao) [inline csds-client/README.md:67]: Client Id

File: csds-client/README.md [hunk 3]
```diff
@@ -64,6 +64,10 @@ Common options are exposed/controlled via command line flags, while control plan
...
+   * If this flag is not specified, all xDS nodes will be returned.
```
Comment (fuqianggao) [inline csds-client/README.md:68]: Client Id

**Review by fuqianggao — APPROVED** [review]

File: csds-client/README.md (hunk 2 above)
Comment (Rainton) [reply csds-client/README.md]: Resolved

File: csds-client/README.md (hunk 3 above)
Comment (Rainton) [reply csds-client/README.md]: Resolved

File: csds-client/client/v2/client_test.go (hunk 1 above)
Comment (Rainton) [reply csds-client/client/v2/client_test.go]: resolved

**Review by fuqianggao — APPROVED** [review]

**Review by htuch — APPROVED** [review]

**Pull request merged**